parachain/pallets/control/src/mock.rs

```diff
@@ -74,135 +171,88 @@ parameter_types! {
-	(
-		[7; 32],
-		MultiLocation {
-			parents: 1,
-			interior: X2(
-				Junction::Parachain(1000),
-				Junction::AccountId32 { network: None, id: [0; 32] },
-			),
-		},
-	),
-	// Case 8: Sibling Chain AccountKey20
-	(
-		[8; 32],
-		MultiLocation {
-			parents: 1,
-			interior: X2(
-				Junction::Parachain(1000),
-				Junction::AccountKey20 { network: None, key: [0; 20] },
-			),
-		},
-	),
-	// Case 9: Bad Multi Locations
-	(
-		[9; 32],
-		MultiLocation {
-			parents: 1,
-			interior: X2(Junction::Parachain(1000), Junction::Parachain(1000)),
-		},
-	),
-	// Case 10: Bad Validate Message
-	([10; 32], MultiLocation { parents: 1, interior: X1(Junction::Parachain(1001)) }),
-	// Case 11: Bad Submit Message
-	([11; 32], MultiLocation { parents: 1, interior: X1(Junction::Parachain(1002)) }),
-];
-
-pub struct EnsureOriginFromTable;
-impl EnsureOrigin<RuntimeOrigin> for EnsureOriginFromTable {
-	type Success = MultiLocation;
-
-	fn try_origin(outer: RuntimeOrigin) -> Result<Self::Success, RuntimeOrigin> {
-		let account = outer.clone().into_signer().ok_or(outer.clone())?;
-
-		// Benchmarking
-		#[cfg(feature = "runtime-benchmarks")]
-		{
-			if account == whitelisted_caller() {
-				return Ok(MultiLocation::new(0, Here))
-			}
-		}
-
-		// test cases
-		let key: [u8; 32] = account.into();
-		for entry in ORIGIN_TABLE {
-			if entry.0 == key {
-				return Ok(entry.1)
-			}
-		}
-		Err(outer)
-	}
-
-	#[cfg(feature = "runtime-benchmarks")]
-	fn try_successful_origin() -> Result<RuntimeOrigin, ()> {
-		Ok(RuntimeOrigin::signed([0u8; 32].into()))
-	}
-}
-
 pub struct MockOutboundQueue;
 impl snowbridge_control::OutboundQueueTrait for MockOutboundQueue {
 	type Ticket = Message;
+	type Balance = u128;
 
-	fn validate(message: &Message) -> Result<Self::Ticket, SubmitError> {
-		Ok(message.clone())
+	fn validate(message: &Message) -> Result<(Self::Ticket, Self::Balance), SubmitError> {
+		Ok((message.clone(), 0))
 	}
 
 	fn submit(_ticket: Self::Ticket) -> Result<MessageHash, SubmitError> {
 		Ok(MessageHash::zero())
 	}
 }
 
-impl snowbridge_control::Config for Test {
+parameter_types! {
+	pub TreasuryAccount: AccountId = PalletId(*b"py/trsry").into_account_truncating();
+	pub Fee: u64 = 1000;
+	pub const RococoNetwork: NetworkId = NetworkId::Rococo;
+}
+
+/// Type for specifying how a `MultiLocation` can be converted into an `AccountId`. This is used
+/// when determining sovereign accounts for asset transacting.
+pub type LocationToAccountId = (
+	// The parent (Relay-chain) origin converts to the parent `AccountId`.
+	ParentIsPreset<AccountId>,
+	// Sibling parachain origins convert to AccountId via the `ParaId::into`.
+	SiblingParachainConvertsVia<Sibling, AccountId>,
+	// Straight up local `AccountId32` origins just alias directly to `AccountId`.
+	AccountId32Aliases<RelayNetwork, AccountId>,
+	// Other nested consensus systems on sibling parachains or relay chain.
+	HashedDescription<AccountId, DescribeFamily<DescribeAllTerminal>>,
+);
```

Comment:
Same config as in bridge_hub and https://github.com/Snowfork/cumulus/pull/67/files#r1346854601